hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/AbstractMapQueryMessageTask.java

```diff
@@ -102,6 +103,11 @@ private void collectResults(Collection<QueryResultEntry> result, List<Future> fu
         for (Future future : futures) {
             QueryResult queryResult = (QueryResult) future.get();
             if (queryResult != null) {
+
+                if (queryResult.isResultLimitExceeded()) {
```

Comment:
I could not test this yet since as far as I saw these tasks are not yet wired in the new client module. But this should hopefully be the right place to fire the exception.